client/src/pages/Question/Question.tsx

```diff
@@ -0,0 +1,136 @@
+          if(timer === 1){
+            clearInterval(a)
+            timeout();
+          }
+          setTime(state => state - 1)
+          timer = timer - 1;
+       },1000)
+
+       if(stopTime){
+          clearInterval(a)
+       }
+
+       return () => {
+          clearInterval(a)
+       }
+    },[stopTime])
+
+    
+    const [isClicked,setIsClicked] = useState<boolean|number>(false);
+
+    const initialState:State = {
+       score:0,
+       currentQuestionNo:0
+    }
+
+    const quizReducer = (state:State,action:Action) => {
+        switch (action.type) {
+            case "INCREMENT_SCORE":
+            return {...state,score:action.payload.score}
+            case "NEXT_QUESTION":
+            return {...state,currentQuestionNo:action.payload.currentQuestionNo}
+            default:
+            return state;
+        }
+    }
+
+    const [{score,currentQuestionNo},dispatch] = useReducer(quizReducer,initialState);
+
+    function handleOption(isRight:boolean,i:number){
+        if(isRight){
+            setIsClicked(true);
+            setStopTime(true);
+            dispatch({type:"INCREMENT_SCORE",payload:{score:score + 1}});
+        }
+        else{
+            setIsClicked(i)
+            setStopTime(true);
+        }
+    }
+
+    const handleNextQues = async() => {
+        setIsClicked(false);
+        setStopTime(false);
+        if(currentQuiz?.questions.length === currentQuestionNo + 1){
+                if(state?.name){
+                    const res = await getLeaderboard(quizID,{name:state?.name,score})
+                    if("leaderboard" in res){
+                        return navigate("/result",{state:{score,leaderboard:res.leaderboard}})
+                    }
+                }
+                return navigate("/result",{state:{score,leaderboard:[]}})
+        }
+        dispatch({type:"NEXT_QUESTION",payload:{currentQuestionNo:currentQuestionNo + 1}});
+        setTime(30);
+    }
+
+    const handleQuit = () => {
+        navigate(-1)
+    }
+
+    return (
+        <div className="question">
+            
+            <div className="display__question__number">{currentQuestionNo + 1}/10</div>
+            <div className="timer__container">
+                <span className={time <=10 ? (time % 2 === 0 ? "timer darkRed" : "timer" )  :"timer"}>{time}</span>
+            </div>
+            <h3 className="question__heading">{currentQuiz?.questions[currentQuestionNo].question}</h3>
+            <div className="options__container">
+              {
+                currentQuiz?.questions[currentQuestionNo].options.map( (option,i) => (
+                    <button 
+                    disabled={isClicked !== false} 
+                    className={
+                        isClicked !== false ? 
+                        ( isClicked === true ? ( option.isRight ? "option right__answer" : "option" ) : ( option.isRight ? "option right__answer" : (i === isClicked ? "option wrong__answer" : "option") ))  
```

Comment:
You need to create a separate function here.Doing so much calculation in return part makes the code ugly

Reply:
done